test/python/transpiler/test_high_level_synthesis.py

```diff
@@ -481,7 +481,7 @@ def test_target_gets_passed_to_plugins(self):
                 [
                     HighLevelSynthesis(
                         hls_config=hls_config,
-                        target=GenericBackendV2(num_qubits=5, basis_gates=["u", "cx"]).target,
+                        target=GenericBackendV2(num_qubits=5, basis_gates=["u", "cx", "id"]).target,
```

Comment:
Just curious: why is this change needed?

Reply:
The `id` gate neither has a definition nor is present in the equivalence library, so HLS complains about it. This manifested only right now since HLS now recursively processes the circuit obtained by synthesizing a high-level-object (allowing a synthesis method to replace a high-level-object by other high-level-objects, somewhat similar to unrolling nested definitions) + one of the mock plugins replaces a custom gate by `id` gates.